tests/solana_tests/rational.rs

```diff
@@ -208,4 +208,32 @@ fn rational() {
             },
         ]
     );
+
+    let mut vm = build_solidity(
+        r#"
+        contract foo {
+            function test1() public returns (bool) {
+                bool x = 1.1 != 1.0;
+                assert(x);
+                assert(1.1 == 1.1);
+                assert(1.2 >= 1.1);
+                assert(1.1 >= 1.1);
+                assert(1.1 <= 1.2);
+                assert(1.1 <= 1.1);
+                assert(1.2 > 1.1);
+                assert(1.1 < 1.2);
+                assert(1.1 > 1.2 || 1.1 < 1.2);
+                assert(1.2 > 1.1 && 1.1 < 1.2);
+                assert(1.2 > 1.1 ? 1.2 < 1.3 : 1.2 > 1.3);
+
+                return 1.1 > 1.0 && 1.1 >= 1.1;
+            }
+        }"#,
+    );
+
+    vm.constructor(&[]);
+
+    let returns = vm.function("test1", &[]).unwrap();
+
+    assert_eq!(returns, BorshToken::Bool(true));
```

Comment:
Rationals are gone after codegen, so there is not much to test. If rationals get left in the cfg emit will panic

Reply:
You call `eval_const_bool` in codegen too. Doesn't constant folding do the same job as this function?

Reply:
I don't think a codegen test would add anything. I don't know what it can test: all the rationals are gone replaced with constants. This can be tested with runtime tests.

Reply:
The `eval_const_bool` function also evaluates integer and boolean expressions inside comparisons. These are not gone in codegen.

Reply:
The new code does not have this change.